elasticapm/utils/disttracing.py

```diff
@@ -208,7 +212,7 @@ def _parse_tracestate(self, tracestate) -> Dict[str, str]:
 
         ret = {}
         try:
-            state = re.search(r"(?:,|^)es=([^,]*)", tracestate).group(1).split(";")
+            state = re.search(r"(?:,|^)\s*es=([^,]*?)\s*(?:,|$)", tracestate).group(1).split(";")
```

Comment:
Is this question mark superfluous? The `*` already makes it optional, yes?

```suggestion
            state = re.search(r"(?:,|^)\s*es=([^,]*)\s*(?:,|$)", tracestate).group(1).split(";")
```

Reply:
That `?` is for a non-greedy match, otherwise, I think (I'd need to test), the `([^,]*)` could slurp up the trailing whitespace rather than leaving it to the `\s*`.

Reply:
```
>>> tracestate = 'foo=bar, es=blah , bing=bang '
>>> re.search(r"(?:,|^)\s*es=([^,]*)\s*(?:,|$)", tracestate).group(1)
'blah '
>>> re.search(r"(?:,|^)\s*es=([^,]*?)\s*(?:,|$)", tracestate).group(1)
'blah'
```

Reply:
Ah, right, thank you. In that case I can undo my "fix" in https://github.com/elastic/apm-agent-python/pull/1728/commits/a609b1a52df7dded2eb7a27720ef223daa875f54

Reply:
Ah, I hadn't see the `.strip()` call. I think the issue with that if this were for **general** tracestate parsing, is that a `value` here can have *leading* whitespace (per https://www.w3.org/TR/trace-context/#value). So it would have to be a `.rstrip()` if kept.   I'm nitpicking though, because IIUC the Python impl *only* ever processing the `es` vendor value and we'll never care about leading whitespace.

Reply:
A possible alternative for parsing this all, if the regex is deemed getting too complex, would be to:
- `str.find('es=')`
- if there was a hit, then `.split(',')` and `.trim()` each entry
- then pick the one(s) starting with 'es='.

Reply:
I think it's fine. But thanks for the thoughtful suggestions.